configure.ac

```diff
@@ -62,6 +62,21 @@ fi
 
 AC_CANONICAL_HOST
 
+AC_MSG_CHECKING([Whether we need POSIX threads])
+AC_COMPILE_IFELSE([
+    AC_LANG_PROGRAM([
+        #if defined(__NetBSD__) || defined(__DragonFly__)
```

Comment:
You could do this without compiling a test program, by checking `host_os`.  See how `WINDOWS` is defined in configure.ac.

Reply:
That is true. I was thinking defensively of the fact that since pthreads are used based on those variables, that is the ultimate source of truth, and this longer configure script would work

I got a little paranoid about it being possible for those variables to be (mis)defined on other systems due to whatever exotic reason, and then we will in fact use POSIX despite what `host_os` said.

But maybe that is being too paranoid :).